File: Baekjoon/BaaarkingDog/0x03_array/11328.cpp
// 11328: Strfry

#include <iostream>
#include <string>
using namespace std;

int main() {
  int N;
  cin >> N;

  while (N--) {
    int alphabet[26] = {0}, cnt = 0;
    string word1, word2;
    cin >> word1 >> word2;

    for (char c : word1) alphabet[c - 'a']++;
    for (char c : word2) alphabet[c - 'a']--;

    for (int a_cnt : alphabet) cnt += (a_cnt > 0 ? a_cnt:-a_cnt);

    cout << (cnt != 0 ? "Impossible":"Possible") << endl;
  }

  return 0;
}

File: Baekjoon/BaaarkingDog/0x03_array/11328.py
# 11328: Strfry

N = int(input())

for _ in range(N):
  alphabet = [0 for _ in range(26)]
  cnt = 0
  word1, word2 = input().split()

  for c in word1: alphabet[ord(c) - ord('a')] += 1
  for c in word2: alphabet[ord(c) - ord('a')] -= 1

  for a_cnt in alphabet: cnt += a_cnt if a_cnt > 0 else -a_cnt

  print("Possible" if cnt == 0 else "Impossible")